### PR TITLE
Refactoring behaviours

### LINKED ISSUE #38: Refactor Behaviour class
Some methods in Behaviour base class are related to a previous way of message management in PADE. Now the message stack is inside the agent, there is no need to this methods remain in Behaviour class.

### PR DESCRIPTION
**Pull request by filipesaraiva**
Removes the message stuffs from Behaviour's base.py and use spaces instead tabs.

### CODE REVIEW

File: pade/behaviours/base.py [hunk 1]
```diff
@@ -6,119 +6,63 @@
...
+    implements the basic methods for scheduled behaviours.
+    '''
+
+    def __init__(self, agent):
+        ''' This method initializes a new instance of BaseBehaviour
+        class and explicitly calls the Behaviour.__init__() method.
+        '''
+        super().__init__(agent)
+        # Lock object (to ensure the mutual exclusion, when needed)
+        self.__lock = None
```
Comment (filipesaraiva) [inline pade/behaviours/base.py:21]: @italocampos, is this attribute and the related methods like `lock` and `add_lock` necessary yet?

File: pade/behaviours/base.py [hunk 2]
```diff
@@ -6,119 +6,63 @@
...
+        pass
+
+
+    def done(self):
+        ''' This is an abstract method that must be overridden in the
+        child classes, dealing with the finish of this behaviour.
+        '''
+        pass
+
+    def wait(self, timeout):
```
Comment (filipesaraiva) [inline pade/behaviours/base.py:36]: @italocampos, this `wait` method works as the `block(int)` method in JADE, right? 🤔

File: pade/behaviours/base.py (hunk 1 above)
Comment (italocampos) [reply pade/behaviours/base.py:21]: Yes. These attributes and proper methods implements the mutual exclusion. Please, keep it on.

File: pade/behaviours/base.py (hunk 2 above)
Comment (italocampos) [reply pade/behaviours/base.py:36]: Right. This doesn't have relation with the behaviours message queue.

**Comment by italocampos** [comment]
@filipesaraiva , you must to remove the `Agent.receive(ACLMessage)` method too. Also, you will need to remove the `SequentialBehaviour.receive(ACLMessage)` in `pade.behaviours.types` module.

File: pade/behaviours/base.py (hunk 1 above)
Comment (filipesaraiva) [reply pade/behaviours/base.py:21]: Sure!

**Review by filipesaraiva — COMMENTED** [review]
@italocampos in https://github.com/grei-ufc/pade/pull/45#issuecomment-650194472:
> @filipesaraiva , you must to remove the `Agent.receive(ACLMessage)` method too.

It was removed in a past commit: https://github.com/grei-ufc/pade/commit/b611f5fe4fe12c875b1405b9c4f657db51ba0161#diff-a61b73126ae6c4b0edda97a330d58194L815

> Also, you will need to remove the `SequentialBehaviour.receive(ACLMessage)` in `pade.behaviours.types` module.

Done!

**Comment by italocampos** [comment]
Ok. Can I proceed the merge?

**Review by italocampos — APPROVED** [review]

**Comment by filipesaraiva** [comment]
@italocampos in https://github.com/grei-ufc/pade/pull/45#issuecomment-650232009:
> Ok. Can I proceed the merge?

Sure, thanks!

**Comment by italocampos** [comment]
I'm accepting this PR.

**Pull request merged**